LandSandBoat/server
# collapse food effect scripts to one, remove dummy effect

**_I affirm:_**

## What does this pull request do?
Long ago a younger me used a dummy effect to implement FoV food. Today I undo my mistake and make it so the core only looks up an item ID if a non zero subType is specified.

Additionally the 2nd commit makes some variable and parameter naming consistent with other parts of the project: status effects have a subType, not a subID, and we used both terms in different places. 

## Steps to test these changes

1. `!additem chunk_of_goblin_chocolate`
2. eat your chocolate
3. !`exec player:PrintToPlayer(string.format("Lizard killer mod: %i", player:getMod(xi.effect.LIZARD_KILLER) ) )` and see that you gained 5 lizard killer. ***note:*** _Remember that if you had lizard killer from gear or job traits, you have a non zero base amount!_
4. wait for that to wear off
5. run that exac command again and see you now have lost that 5 lizard killer. 
6. get dried meat from a field manual. you can use `!addcurrency field_point 5000 <me>` if you need tabs to spend on it.
7. check status screen and see that you gained the 4 str it gives.
8. wait for it to wear off. see that the str was removed.
